smack-integration-test/src/main/java/org/jivesoftware/smackx/usertune/UserTuneIntegrationTest.java

```diff
@@ -0,0 +1,97 @@
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.usertune;
+
+import java.net.URI;
+
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smackx.usertune.element.UserTuneElement;
+
+import org.igniterealtime.smack.inttest.AbstractSmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTestEnvironment;
+import org.igniterealtime.smack.inttest.util.IntegrationTestRosterUtil;
+import org.igniterealtime.smack.inttest.util.SimpleResultSyncPoint;
+import org.junit.AfterClass;
+
+import org.jxmpp.jid.BareJid;
+
+public class UserTuneIntegrationTest extends AbstractSmackIntegrationTest {
+
+    private final UserTuneManager utm1;
+    private final UserTuneManager utm2;
+
+    public UserTuneIntegrationTest(SmackIntegrationTestEnvironment<?> environment) {
+        super(environment);
+        utm1 = UserTuneManager.getInstanceFor(conOne);
+        utm2 = UserTuneManager.getInstanceFor(conTwo);
+    }
+
+    @SmackIntegrationTest
+    public void test() throws Exception {
+        URI uri = new URI("http://www.yesworld.com/lyrics/Fragile.html#9");
+        UserTuneElement.Builder builder = UserTuneElement.getBuilder();
+        UserTuneElement userTuneElement1 = builder.setArtist("Yes")
+                                                  .setLength(686)
+                                                  .setRating(8)
+                                                  .setSource("Yessongs")
+                                                  .setTitle("Heart of the Sunrise")
+                                                  .setTrack(3)
+                                                  .setUri(uri)
+                                                  .build();
+
+        IntegrationTestRosterUtil.ensureBothAccountsAreSubscribedToEachOther(conOne, conTwo, timeout);
+
+        final SimpleResultSyncPoint userTuneReceived = new SimpleResultSyncPoint();
+
+        final UserTuneListener userTuneListener = new UserTuneListener() {
+            @Override
+            public void onUserTuneUpdated(BareJid jid, Message message, UserTuneElement userTuneElement) {
+                if (isEqual(userTuneElement,userTuneElement1)) {
+                    userTuneReceived.signal();
+                }
+            }
+        };
+
+        utm2.addUserTuneListener(userTuneListener);
+
+        try {
+            utm1.publishUserTune(userTuneElement1);
+            userTuneReceived.waitForResult(timeout);
+        } finally {
+            utm2.removeUserTuneListener(userTuneListener);
+        }
+    }
+
+    private boolean isEqual(UserTuneElement usertuneElement,UserTuneElement userTuneElement1) {
+        if (usertuneElement.toXML().toString().contentEquals(userTuneElement1.toXML().toString())) {
```

Comment:
This can be simplified to `return usertuneElement.toXML().toString().contentEquals(userTuneElement1.toXML().toString();`